src/clarity-angular/layout/tabs/tab.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {Component, ContentChild, forwardRef, Inject, InjectionToken} from "@angular/core";
+import {TabLink} from "./tab-link";
+import {TabContent} from "./tab-content";
+import {IfActiveService} from "../../utils/conditional/if-active.service";
+
+let tabCounter = 0;
+
+export const IF_ACTIVE_ID = new InjectionToken<number>("IF_ACTIVE_ID");
+
+export function tokenFactory() {
+    return ++tabCounter;
+}
+
+export const IF_ACTIVE_ID_PROVIDER = {
+    provide: IF_ACTIVE_ID,
+    useFactory: tokenFactory
+};
+
+@Component({
+    selector: "clr-tab",
+    template: `
+        <ng-content select="clr-tab-link"></ng-content>
+        <ng-content select="clr-tab-content"></ng-content>
+    `,
+    providers: [ IF_ACTIVE_ID_PROVIDER ]
+})
+export class Tab {
+    @ContentChild(forwardRef(() => TabLink)) tabLink: TabLink;
+    @ContentChild(forwardRef(() => TabContent)) tabContent: TabContent;
+
+    constructor(public ifActiveService: IfActiveService, @Inject(forwardRef(() => IF_ACTIVE_ID)) public self: any) {
+    }
+
+    ngAfterContentChecked() {
+        // the tabContent is only there if this tab is active
+        if (this.tabLink && this.tabContent) {
+            this.tabLink.ariaControls = this.tabContent.id;
+            this.tabContent.ariaLabelledBy = this.tabLink.id;
+        }
```

Comment:
Getting rid of the parent's `ngAfterContentInit` got rid of the chocolate error. Let me know if you think that's insufficient and I can explore this more.